src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -395,6 +409,175 @@ bool LFUDAPolicy::erase(const DoutPrefixProvider* dpp, const std::string& key, o
+	rgw_user c_rgw_user = it->second->user;
+
+	size_t pos = 0;
+	std::string delimiter = "_";
+	while ((pos = name.find(delimiter)) != std::string::npos) {
+	  if (count == 0){
+	    b_name = name.substr(0, pos);
+    	    name.erase(0, pos + delimiter.length());
+	  }
+	  count ++;
+	}
+	key = name;
+
+	//writing data to the backend
+	//we need to create an atomic_writer
+ 	rgw_obj_key c_obj_key = rgw_obj_key(key); 		
+	std::unique_ptr<rgw::sal::User> c_user = driver->get_user(c_rgw_user);
+
+	std::unique_ptr<rgw::sal::Bucket> c_bucket;
+        rgw_bucket c_rgw_bucket = rgw_bucket(c_rgw_user.tenant, b_name, "");
+
+	RGWBucketInfo c_bucketinfo;
+	c_bucketinfo.bucket = c_rgw_bucket;
+	c_bucketinfo.owner = c_rgw_user;
+	
+	
+    	ldpp_dout(dpp, 10) << __func__ << " : "  << __LINE__ << ": bucket name is " << b_name << dendl;
+    	int ret = driver->load_bucket(dpp, c_rgw_bucket, &c_bucket, null_yield);
+	if (ret < 0) {
+      	  ldpp_dout(dpp, 10) << __func__ << "(): load_bucket() returned ret=" << ret << dendl;
+      	  break;
+        }
+    	ldpp_dout(dpp, 10) << __func__ << " : "  << __LINE__ << ": bucket name is " << c_bucket->get_name() << dendl;
+	/*
+    	c_bucketinfo = bucket->get_info();
+    	ldpp_dout(dpp, 10) << __func__ << " : "  << __LINE__ << ": bucket name is " << b_name << dendl;
+	c_bucket = driver->get_bucket(c_bucketinfo);
+    	ldpp_dout(dpp, 10) << __func__ << " : "  << __LINE__ << ": bucket name is " << c_bucket->get_name() << dendl;
+	*/
+	/*
+ 	int ret = driver->get_bucket(dpp, nullptr, c_rgw_bucket, &c_bucket, null_yield);
+	if (ret < 0){
+    		ldpp_dout(dpp, 10) << __func__ << " : "  << __LINE__ << " cleaning get_bucket() failed for Bucket: " << b_name << dendl;
+		continue;
+	}
+	*/
+
+	std::unique_ptr<rgw::sal::Object> c_obj = c_bucket->get_object(c_obj_key);
+
+	std::unique_ptr<rgw::sal::Writer> processor =  driver->get_atomic_writer(dpp,
+				  null_yield,
+				  c_obj.get(),
+				  c_user->get_id(),
+				  NULL,
+				  0,
+				  "");
+
+  	int op_ret = processor->prepare(null_yield);
+  	if (op_ret < 0) {
+    	  ldpp_dout(dpp, 20) << "processor->prepare() returned ret=" << op_ret << dendl;
+    	  break;
+  	}
+
+	std::string prefix = b_name+"_"+key;
+	off_t lst = it->second->size;
+  	off_t fst = 0;
+  	off_t ofs = 0;
+
+	
+  	rgw::sal::DataProcessor *filter = processor.get();
+	do {
+    	  ceph::bufferlist data;
+    	  if (fst >= lst){
+      	    break;
+    	  }
+    	  off_t cur_lst = std::min<off_t>(fst + cct->_conf->rgw_max_chunk_size, lst);
+    	  std::string oid_in_cache = "D_" + prefix + "_" + std::to_string(fst) + "_" + std::to_string(cur_lst);  	  
+    	  std::string new_oid_in_cache = prefix + "_" + std::to_string(fst) + "_" + std::to_string(cur_lst);
+      	  ldpp_dout(dpp, 20) << __func__  << ": " << __LINE__ << ": oid_in_cache is: " << oid_in_cache << dendl;
+    	  cacheDriver->get(dpp, oid_in_cache, fst, cur_lst, data, obj_attrs, null_yield);
```

Comment:
@mosayyebzadeh : please use `yield_context` instead of `null_yield`.

Reply:
I am not sure how to get a valid yield_context here.

Reply:
ok, yes we do not have access to the yield_context in the `init` method too, for the time being null yield is ok, we can come back to this later also.